test/unit/handlebars.cpp

```diff
@@ -0,0 +1,460 @@
+                frame.set("..", cb.context());
+                frame.set("@key", i);
+                frame.set("@first", i == 0);
+                frame.set("@last", i == items.size() - 1);
+                frame.set("@index", i);
+                out += "<li>" + cb.fn(frame) + "</li>";
+            }
+            return out + "</ul>";
+        }
+        return cb.inverse();
+    });
+
+    hbs.registerHelper("isdefined", [](dom::Array const& args) -> dom::Value {
+        if (args.empty()) {
+            return "isdefined helper requires at least one argument";
+        }
+        // There's no distinction between null and undefined in mrdox::dom
+        return !args[0].isNull();
+    });
+
+    hbs.registerHelper("helperMissing", [](
+        dom::Array const& args,
+        HandlebarsCallback const& cb) -> dom::Value {
+        std::string out;
+        OutputRef os(out);
+        os << "Missing: ";
+        os << cb.name;
+        os << "(";
+        for (std::size_t i = 0; i < args.size(); ++i) {
+            if (i != 0) {
+                os << ", ";
+            }
+            os << args[i];
+        }
+        os << ")";
+        return out;
+    });
+
+    hbs.registerHelper("blockHelperMissing", [](
+        dom::Array const& args,
+        HandlebarsCallback const& cb) -> dom::Value {
+        std::string out;
+        OutputRef os(out);
+        os << "Helper '";
+        os << cb.name;
+        os << "' not found. Printing block: ";
+        os << cb.fn();
+        return out;
+    });
+
+    /////////////////////////////////////////////////////////////////
+    // Register partials
+    /////////////////////////////////////////////////////////////////
+    // From files
+    for (auto partial_path: partial_paths) {
+        auto partial_text_r = files::getFileText(partial_path);
+        REQUIRE(partial_text_r);
+        std::string_view filename = files::getFileName(partial_path);
+        auto pos = filename.find('.');
+        if (pos != std::string_view::npos) {
+            filename = filename.substr(0, pos);
+        }
+        hbs.registerPartial(filename, *partial_text_r);
+    }
+
+    // Dynamic partial helpers
+    hbs.registerHelper("whichPartial", []() -> dom::Value {
+        return "dynamicPartial";
+    });
+
+    // Literal partials
+    hbs.registerPartial("dynamicPartial", "Dynamo!");
+    hbs.registerPartial("lookupMyPartial", "Found!");
+    hbs.registerPartial("myPartialContext", "{{information}}");
+    hbs.registerPartial("myPartialParam", "The result is {{parameter}}");
+    hbs.registerPartial("myPartialParam2", "{{prefix}}, {{firstname}} {{lastname}}");
+    hbs.registerPartial("layoutTemplate", "Site Content {{> @partial-block }}");
+    hbs.registerPartial("pageLayout", "<div class=\"nav\">\n  {{> nav}}\n</div>\n<div class=\"content\">\n  {{> content}}\n</div>");
+
+    /////////////////////////////////////////////////////////////////
+    // Render and diff
```

Comment:
Oh... okay, so you are doing the render and then diff with the source data but where is the folder with all the triplets of files (input file, template file, expected output file)?

Reply:
`test/unit/fixtures`